### DISCUSSION TITLE
How to use for `compileOnly` by dependsOn ?

### DISCUSSION THREAD

**Discussion by laglangyue** [sbt/sbt#7476] [Q&A]
My project `core` depensOn `common`, common has some `lib`, but `core` don't need them.
for gradle
```shell
  api(project(":common")) {
    transitive = false
  }
```
for sbt,
I try use  `dependsOn(ndiCommon % Compile)`, it will carry libs for core,
then try use `dependsOn(ndiCommon % Provide)`, it will lost `common jar`

BTW, I use sbt-pack to do the experiment.
What can I do

**Comment by adpi2** [comment]
I don't think there is such thing as `transitve = false` in sbt for projects. The dependencies of a project are available for other projects that depend on it.

**Reply by laglangyue** [reply]
Indeed, common depends on spark, but the core does not depend on it. It is possible to separate the common part into a new module, which both the common and core depend on.
